statsmodels/stats/meta_analysis.py

```diff
@@ -0,0 +1,243 @@
+                  self.ci_low_smd_fe, self.ci_upp_smd_fe, 1, np.nan]]
+        res_re = [[self.smd_effect_re, self.sd_smd_w_re,
+                  self.ci_low_smd_re, self.ci_upp_smd_re, np.nan, 1]]
+
+        res = np.concatenate([res, res_fe, res_re], axis=0)
+        column_names = ['smd', "sd_smd", "ci_low", "ci_upp", "w_fe", "w_re"]
+        return res, column_names
+
+    def summary_frame(self):
+        labels = list(self.row_names) + ["fixed effect", "random effect"]
+        res, col_names = self.summary_array()
+        results = pd.DataFrame(res, index=labels, columns=col_names)
+        return results
+
+
+def effectsize_smd(mean2, sd2, nobs2, mean1, sd1, nobs1, row_names=None,
+                   alpha=0.05):
+    """effect sizes for mean difference for use in meta-analysis
+
+    mean2, sd2, nobs2 are for treatment
+    mean1, sd1, nobs1 are for control
+
+    Effect sizes are computed for the mean difference ``mean2 - mean1``
+
+    This does not have option yet.
+    It uses standardized mean difference with bias correction as effect size.
+
+    This currently does not use np.asarray, all computations are possible in
+    pandas.
+
+    References
+    ----------
+    Borenstein, Michael. 2009. Introduction to Meta-Analysis.
+        Chichester: Wiley.
+
+    Chen, Ding-Geng, and Karl E. Peace. 2013. Applied Meta-Analysis with R.
+        Chapman & Hall/CRC Biostatistics Series.
+        Boca Raton: CRC Press/Taylor & Francis Group.
+
+
+    """
+    k = len(mean2)
+    if row_names is None:
+        row_names = list(range(k))
+    crit = stats.norm.isf(alpha / 2)
+
+    var_diff_uneq = sd2**2 / nobs2 + sd1**2 / nobs1
+    var_diff = (sd2**2 * (nobs2 - 1) +
+                sd1**2 * (nobs2 - 1)) /  (nobs2 + nobs1 - 2)
+    sd_diff = np.sqrt(var_diff)
+    nobs = nobs2 + nobs1
+    bias_correction = 1 - 3 / (4 * nobs - 9)
+    smd = (mean2 - mean1) / sd_diff
+    smd_bc = bias_correction * smd
+    var_smdbc = nobs / nobs2 / nobs1  + smd_bc**2 / 2 / (nobs - 3.94)
+    return smd_bc, var_smdbc
+
+
+def combine_effects(effect, variance, row_names=None, alpha=0.05):
+    """combining effect sizes for effect sizes using meta-analysis
+
+
+
+    This does not have option yet.
+    It uses standardized mean difference with bias correction as effect size.
+
+    This currently does not use np.asarray, all computations are possible in
+    pandas.
+
+    References
+    ----------
+    Borenstein, Michael. 2009. Introduction to Meta-Analysis.
+        Chichester: Wiley.
+
+    Chen, Ding-Geng, and Karl E. Peace. 2013. Applied Meta-Analysis with R.
+        Chapman & Hall/CRC Biostatistics Series.
+        Boca Raton: CRC Press/Taylor & Francis Group.
+
+
+    """
+    k = len(effect)
+    if row_names is None:
+        row_names = list(range(k))
+    crit = stats.norm.isf(alpha / 2)
+
+    # alias for initial version
+    smd_bc = effect
+    var_smdbc = variance
+    sd_smdbc = np.sqrt(var_smdbc)
```

Comment:
names with `smd` for standardized mean difference are still from the initial special case.
we need more general names, eg. `eff` 
`smd_effect` is the overall mean estimate, need better name, `mean_effect`, `mean_eff`, ... ?